lib/galaxy/tools/actions/__init__.py

```diff
@@ -97,13 +97,13 @@ def process_dataset(data, formats=None):
                 # fetch dataset details for this input from the database.
                 if collection_info and collection_info.is_mapped_over(input_name):
                     action_tuples = collection_info.map_over_action_tuples(input_name)
-                    if not trans.app.security_agent.can_access_datasets(current_user_roles, action_tuples):
+                    if not trans.user_is_admin and not trans.app.security_agent.can_access_datasets(current_user_roles, action_tuples):
```

Comment:
(disclaimer, haven't touched back-end things in a while) Wouldn't it make sense to just add this to `trans.app.security_agent.can_access_datasets`? If the user is an admin, then `can_access_datasets` should immediately return true?

Reply:
That was exactly the first idea, but unfortunately `can_access_datasets` only deals with user roles, and apparently, we can only know if the user is an admin through the `trans` object or by checking the galaxy config in the security agent.
Then in the backend chat, the idea of using `is_permitted` popped up, which already does something like that, but it required to use of other managers so, all of the options required significant changes or modifying existing interfaces and I thought that they should be part of a proper redesign and not part of this fix.

But maybe there is an easier or simpler way of doing it, I just didn't figure it out :sweat_smile:

Reply:
Ah I missed there was out of band discussion/review, feel free to ignore it then :) 
> thought that they should be part of a proper redesign and not part of this fix.
sounds like a good plan